wordpress-mobile/WordPress-Android
# [Do Not Merge] [Dependency Updates] Update `firebaseConfigVersion` to 21.4.0

Parent #17567

This PR update `firebaseConfigVersion` to [21.4.0](https://firebase.google.com/support/release-notes/android#remote-config_v21-4-0).

Also, as part of this update the below transitive dependencies were added:

- On the `WordPress` module (5c1eebab5b2ee4f09576f05a84cbc5559a2f63ed):
    - `com.google.firebase:firebase-iid`

-----

PS: @zwarm @AjeshRPai I added you as the main reviewers, not so randomly (*), since I just wanted someone from the WordPress team to be aware of and sign-off on that change for WPAndroid. I also added the @wordpress-mobile/apps-infrastructure team, but this in done only for monitoring purposes, as such, I am not expecting any active review from that team. Thus, feel free to merge this PR if you deem so.

(*): The not so randomly happened because I wanted to get in touch with you to understand more on how you would test this update, the fact that you were involved with this #17497 PR, and the fact that we are both involved with the feature that related to this #13980 PR (aka `Scan` and `Threat Details`).

-----

Refactor List:

1. [Move codeable get free estimate url remote config to wrapper](https://github.com/wordpress-mobile/WordPress-Android/commit/3698e549d3686eaac39d3c94e5ad9771ed06ab84)

-----

## To test:

1. See the dependency tree diff result and verify correctness.
2. Thoroughly smoke test any Firebase remote config functionality on both, the WordPress and Jetpack apps, and see if everything is working as expected.
3. In addition to the above smoke test, you can expand the below and follow the inner and more explicitly test steps within:

<details>
    <summary>1. Deep Linking Functionality [DeepLinkingIntentReceiverActivity.java + DeepLinkingIntentReceiverViewModel.kt + DeepLinkOpenWebLinksWithJetpackHelper.kt]</summary>

ℹ️ This test applies to both, the `WordPress` and `Jetpack` apps.
⁉️ See testing instructions on this #17497 PR that introduced this remote config.
</details>

<details>
    <summary>2. Threat Details Screen [ThreatDetailsFragment.kt + ThreatDetailsViewModel.kt + ThreatDetailsNavigationEvents.kt]</summary>

ℹ️ This test applies to both, the `WordPress` and `Jetpack` apps.
⁉️ See testing instructions on this #13980 PR that introduced this remote config.
</details>

<details>
    <summary>3. Crash Logging Functionality [CrashLoggingModule.java + WPCrashLoggingDataProvider.kt + WPPerformanceMonitoringConfig.kt]️</summary>

ℹ️ This test applies to both, the `WordPress` and `Jetpack` apps.
⁉️ See testing instructions on this #18496 PR that introduced this remote config.
</details>

FYI: I also created this temporal Firebase related [deps_update_firebase_config_to_21_4_0](https://console.firebase.google.com/project/api-project-108380595987/config?query=deps_update_firebase_config_to_21_4_0) remote config to try and test this update, or Firebase Remote Config in general, as I just couldn't and that kind of raised a question, that is, whether Firebase Remote Config works at all. 🤔

-----

## Merge instructions

- [x] @zwarm let's figure out how to properly test the Firebase Remote Config functionality, using this temporal Firebase related [deps_update_firebase_config_to_21_4_0](https://console.firebase.google.com/project/api-project-108380595987/config?query=deps_update_firebase_config_to_21_4_0) remote config, or any other way. For some reason I couldn't and I was keep getting an empty string value (`""`). The same apply for this `2. Threat Details Screen` test above. Let's discuss as I might be missing something trivial here... 🤷 -> [Related Comment](https://github.com/wordpress-mobile/WordPress-Android/pull/18513#issuecomment-1564158158)

-----

## Regression Notes

1. Potential unintended areas of impact

    - Potential breakage or misbehaviour on any or all Firebase related remote config functionalities.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR submission checklist:


UI Changes testing checklist:
